.github/workflows/docker_linux_x86_64.yml

```diff
@@ -43,7 +43,7 @@ jobs:
     steps:
 
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
```

Comment:
@vincentmr  what would be the advantage of this change? is this required by any other task?

Reply:
Just some maintenance to remove a bunch of warnings.

Reply:
Node.js 16 actions are deprecated, and several actions are warning about it.